### PR TITLE
Add native FileEditor

### PR DESCRIPTION
**Pull request by vanwagonet**
[run-nightly]
refs: MBL-14705
affects: Student, Teacher
release note: none

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Student.ipa</summary>
  <a href="https://app.bitrise.io/artifact/49163806/p/340bca90010b413a03bd23134b47114f">
    
  </a>
</details>
<details><summary>Teacher.ipa</summary>
  <a href="https://app.bitrise.io/artifact/49163814/p/763d6f842f7b0a6de07c3616e79b304c">
    
  </a>
</details>
<details><summary>Parent.ipa</summary>
  <a href="https://app.bitrise.io/artifact/49163784/p/70fceca8775392cb6bfe062532fb1533">
    
  </a>
</details>

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 93.1% | 93.17% | -0.07%

[MBL-14705](https://instructure.atlassian.net/browse/MBL-14705)

#### Affected Apps: Student, Teacher
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/Core/Files/FileList/FileListViewController.swift [hunk 1]
```diff
@@ -422,7 +421,7 @@ class FileListCell: UITableViewCell {
             )
             return
         }
-        if let url = item?.file?.thumbnailURL {
+        if let url = item?.file?.thumbnailURL, let c = item?.file?.createdAt, Clock.now.timeIntervalSince(c) > 3600 {
```
Comment (armstrongnate) [inline Core/Core/Files/FileList/FileListViewController.swift:424]: Nice 😆

**Review by armstrongnate — COMMENTED** [review]
Code looks great to me. I like the `load` and `loaded` pattern. OptionalDatePicker is nice and the EditorForm components are great. Well done. QA'ing next.

**Review by armstrongnate — CHANGES_REQUESTED** [review]
Can we add the ability to set the time with the date picker? Right now we display the time (it's always the current hour and minutes). The default time might not be what they want and it's a little confusing that we display it but they can't change it.

Also, the errors that are displayed when the 'Available From' time is after 'Available To' is the same error in both places. Is that expected? Maybe one should say 'Available From should come before Available To' and the other say 'Available To should come after Available From'.

**Comment by armstrongnate** [comment]
Oh wait, I didn't even have the feature flag enabled 🤦 . Well, I guess I found a few issues with the current form. Wow.

File: Core/Core/Files/FileEditor/FileEditorView.swift [hunk 2]
```diff
@@ -84,31 +79,275 @@ public struct FileEditorView: View {
...
+                    .identifier("FileEditor.accessButton")
+
+                if access == .scheduled {
+                    Divider()
+                    OptionalDatePicker(selection: $unlockAt, max: lockAt, initial: Clock.now.startOfDay()) {
+                        Text("Available from")
+                    }
+                        .identifier("FileEditor.unlockAtButton")
+                    Divider()
+                    OptionalDatePicker(selection: $lockAt, min: unlockAt, initial: Clock.now.endOfDay()) {
```
Comment (armstrongnate) [inline Core/Core/Files/FileEditor/FileEditorView.swift:134]: Oh the use of min and max here is so nice!!

**Review by armstrongnate — CHANGES_REQUESTED** [review]
Okay for real this time:

1. Getting a crash when I change the name of a file or folder then tap Done
2. Add new file > Edit the new file > Change Access to Publish > Tap Done > Nothing happens (the modal should dismiss)
3. For the date pickers, in order to use the default date I have to change the date to something else then change it back. That's a little weird. Can we select the default date once the date picker is shown and not require a change? This is how the RN version worked.

File: rn/Teacher/ios/TeacherUITests/Files/FilesUITests.swift [hunk 3]
```diff
@@ -95,13 +97,13 @@ class FilesUITests: MiniCanvasUITestCase {
         FileList.file(index: 0).tap()
         FileDetails.editButton.tap()
 
-        FileEditor.copyrightField.typeText("me")
-        let picker = app.pickerWheels.firstElement
-        FileEditor.justificationButton.tapUntil { picker.exists }
-        picker.rawElement.adjust(toPickerWheelValue: "Public Domain File")
+        // FileEditor.copyrightField.typeText("me") // Can't type in SwiftUI TextField?!
```
Comment (armstrongnate) [inline rn/Teacher/ios/TeacherUITests/Files/FilesUITests.swift:100]: Oh dear.

**Review by armstrongnate — CHANGES_REQUESTED** [review]
I am still getting a crash every time I edit the name of a file or folder and tap Done. It only happens on iPad. I can't think of why that would be. Hopefully you can repro in debug mode 🤞

File: rn/Teacher/ios/TeacherUITests/Files/FilesUITests.swift (hunk 3 above)
Comment (vanwagonet) [reply rn/Teacher/ios/TeacherUITests/Files/FilesUITests.swift:100]: I hope we figure something out that works later, but I figured that wasn't crucial for this test.

**Review by armstrongnate — APPROVED** [review]
QA +1

**Pull request merged**